content/en/blog/_posts/2022-07-27-pod-os-field/index.md

```diff
@@ -0,0 +1,50 @@
+# Identify Pod Operating System authoritatively during pod admission time
+---
+layout: blog
+title: "Identifying pod operating system authoritatively during pod admission time"
+date: 2022-07-27
+slug: Identifying pod operating system authoritatively during pod admission time
+---
+
+**Authors:** Ravi Gudimetla (Apple)
+
+This blog describes how to identify pod's operating system authoritatively using the `OS` field in the pod spec and how it 
+helps in securing the kubernetes clusters while ensuring that the workloads can run smoothly.  
+
+## What problems does this solve?
+
+Today, the `kube-apiserver` and `kubelet` while admitting pods have no notion of the OS on which the pod can run. 
+As a result, the end user of the kubernetes cluster can set Linux specific security constraints onto Windows pods or vice-versa. 
+These security constraints are passed on to the underlying container runtime from kubelet and since these security constraints 
+are specific to OS, the containers may fail to come up.
+This problem worsens if we have admission plugins or webhooks like `PodSecurity` in our kubernetes clusters which adds the
+security constraints to the pods based on the security profile we choose.
+While the `kubelet` can identify the OS on which it runs and can strip certain security constraints before passing
+onto the container runtime, it'd be better to identify the OS early in the life-cycle of pod and use the 
+identification mechanism consistently across all kubernetes components including `kubelet`. 
+With that in mind, `OS` field has been added to the pod spec in 1.23 release of kubernetes as an 
+alpha feature and it graduated to stable in 1.25.`PodSecurity` admission plugin has been updated to use the `OS` field. 
+
+
+## How does it work?
+A new field called `OS` has been added to the pod spec. Every object in kubernetes is validated before it 
+gets persisted to etcd during API admission time, including pod object. Using the `OS` field in pod spec,
+we introduced new validation which forbids Linux specific constraints to be set on Windows pods and vice-versa.
+
+The pod security standards `restricted` profile has been updated to ignore the following policies 
+- AllowPrivilegeEscalation
+- Capabilities
+- SeccompProfile
+when `pod.Spec.OS.Name` is set to `Windows`
+
+
+## How do I use it?
+By setting `pod.Spec.OS.Name=Windows` or `pod.Spec.OS.Name=Linux`
+
+## How can I learn more?
+KEP: https://github.com/kubernetes/enhancements/issues/2802
+API Changes: https://github.com/kubernetes/kubernetes/pull/104693
```

Comment:
Once https://kubernetes.io/docs/concepts/workloads/pods/ mentions the new field (see https://github.com/kubernetes/website/pull/35590), please link there.